integration_tests/src/main/python/join_test.py

```diff
@@ -210,6 +210,27 @@ def do_join(spark):
         return left.join(right, left.key == right.r_key, join_type)
     assert_gpu_and_cpu_are_equal_collect(do_join, conf=_hash_join_conf)
 
+# Test GpuBroadcastShuffledHashJoin with condition is converted to GpuBroadcastNestedLoopJoin
+@validate_execs_in_gpu_plan('GpuBroadcastNestedLoopJoinExec')
+@ignore_order(local=True)
+@pytest.mark.parametrize('data_gen', ast_gen, ids=idfn)
+@pytest.mark.parametrize('join_type', ['Left', 'LeftSemi', 'LeftAnti'], ids=idfn)
+def test_right_broadcast_hash_join_ast_override(data_gen, join_type):
+    def do_join(spark):
+        left, right = create_ridealong_df(spark, short_gen, data_gen, 500, 50)
+        return left.join(broadcast(right), (left.b >= right.r_b), join_type)
```

Comment:
I was trying to test the conversion from GpuBroadcastHashJoin to GpuBroadcastNestedLoopJoin.  I originally had versions of these with no condition, and they were definitely using GpuBroadcastHashJoin, which I verified with: `@validate_execs_in_gpu_plan('GpuBroadcastHashJoinExec')`.  I'm not sure why I deleted those versions.  I will add them back in my next commit so you can see what I have.  They were modeled after `test_hash_join_ridealong`, and use the `_hash_join_conf`.

Reply:
The join condition will need to be on two key columns.  One will be testing for equality while the other needs to test for an inequality.  Therefore `create_ridealong_df` is not appropriate for this, since it only creates one key column that is joinable and another that is not (it's a nested type).